pkg/feeder/feeder.go

```diff
@@ -53,14 +54,23 @@ func NewClient(baseURL, baseAPI string, client *HttpClient) *Client {
 	// retry mechanism for do requests
 	retryFuncForDoReq := func(req *http.Request, httpClient HttpClient, err error) (*http.Response, error) {
 		var res *http.Response
-		for i := 0; err != nil && i < 2; i++ {
-			time.Sleep(time.Second * 10)
+		wait := 5 * time.Second
+		for i := 0; err != nil && i < 10; i++ {
 			res, err = httpClient.Do(req)
+			if err == nil {
+				return res, nil
+			}
+			fmt.Println("Waiting to do again a request:", wait.Seconds())
+			time.Sleep(wait)
+			wait = wait * 2
 		}
```

Comment:
I suggest using the `err != nil` comparison here (more Go style), and using the `break` instruction to stop the `for` instead of an `err != nil` comparison in the `for` check condition. With these changes the `time.Sleep(wait)` can be moved into the `if err != nil` statement, and then the code will be more readable.

Reply:
I introduced other vars there, could you check again? should be solved.